store/v2/storage/sqlite/batch.go

```diff
@@ -74,29 +75,37 @@ func (b *Batch) Delete(storeKey, key []byte) error {
 	return nil
 }
 
-func (b *Batch) Write() error {
-	_, err := b.tx.Exec(reservedUpsertStmt, reservedStoreKey, keyLatestHeight, b.version, 0, b.version)
+func (b *Batch) Write() (err error) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	err = b.db.Begin()
```

Comment:
is there an overhead to calling this on every write?

Reply:
Yes and that overheard already existed, SQLite is single writer so `mattn/go-sqlite3` was synchronizing on a mutex too (or something like it, I haven't read the whole library)